.github/workflows/bump_torch_mlir.yml

```diff
@@ -0,0 +1,67 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# Automatically bumps the torch-mlir submodule to match shark-infra/torch-mlir:main.
+# Manually triggered for now, but can be scheduled in the future.
+# This typically takes ~90s to run.
+name: Bump submodule - shark-infra/torch-mlir
+
+on:
+  # TODO: Uncomment the following lines to enable the schedule when this action is stable & useful enough
+  # schedule:
+  #   - cron: '0 0 * * 0,4'  # Run at 00:00 on Sunday and Thursday
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths:
+      - .github/workflows/bump_torch_mlir.yml
```

Comment:
Do we want this to run on _pushes_ to `main` or on pull requests affecting the workflow that target `main`? I'd think the latter?